capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -34,19 +47,69 @@ export class WikiMediaComponent implements OnChanges, OnInit {
   }
 
   getResults(queryString: string) {
-    this.wikiService.search(queryString)
+    this.wikiService.search(queryString, 'en', 'History')
       .subscribe((result: WikiServiceResult) => {
         if (result.history) {
           this.body = result.history;
           this.history = result.history;
           this.urls = result.furtherReading;
           this.title = result.title;
           this.loading = false;
+          this.langlinks = result.langlinks;
+          const language = this.langlinks.get('ca');
+          this.prevQuery = language.get('searchQuery');
+          this.prevWordForHistory = 'History';
+          this.prevPreFix = 'en';
         }
         else {
           this.error = 'API did not return a valid response.';
           console.error('API did not return a valid response.');
         }
     });
   }
+
+  changeLanguage(queryString: string, languagePrefix: string, wordForHistory: string) {
+    this.wikiService.searchNewLang(queryString, languagePrefix, wordForHistory)
+      .subscribe((result: WikiServiceResult) => {
+        this.history = result.history;
+        if (!(this.history) || (this.history.includes("CPU time usage:"))){
+          this.loading = true;
+          this.error = "Unfortunately, the language you requested does not have an available tranlation for this page. Please select a different language.";
+        }
+        else {
+          this.body = this.history;
+          this.loading = false;
+          this.title = result.title;
+          this.urls = result.furtherReading;
+          if(result.langlinks && result.langlinks.size > 0) {
+              this.langlinks = result.langlinks;
+          }
+        }
+    });
+  }
+
+  onChangeLanguage(prefix: string) {
+    if (prefix == this.prevPreFix) {
+      this.changeLanguage(this.prevQuery, this.prevPreFix, this.prevWordForHistory);
+    }
+    else if (prefix == 'en' && !(this.langlinks.get('en'))) {
+      this.prevPreFix = prefix;
+      const language = this.langlinks.get(prefix);
+      this.query = language.get('searchQuery');
+      this.prevQuery = this.query;
+      this.language = 'English';
```

Comment:
It's a little confusing to have a local variable language and a class variable language. Additionally, didn't you set the class variable language to equal the Languages object and now it's being assigned a string?